include/deal.II/base/table.h

```diff
@@ -404,6 +404,17 @@ class TableBase : public Subscriptor
 {
 public:
   /**
+   * Declare standard types used in all containers.
+   */
+  typedef T                                                 value_type;
+  typedef value_type                                       *pointer;
+  typedef const value_type                                 *const_pointer;
+  typedef value_type                                       *iterator;
+  typedef const value_type                                 *const_iterator;
```

Comment:
Well, but if you introduce iterator types, you also need to have iterators (which we don't). I think you should only introduce those typedefs that would actually be useful. Specifically, this would be `value_type`.